src/components/layers-panel/arcgis-import-panel/arcgis-import-panel.tsx

```diff
@@ -0,0 +1,302 @@
+`;
+
+const TableRow = styled.tr<{ checked: boolean }>`
+  background: transparent;
+  cursor: pointer;
+  border: 0;
+  border-style: solid;
+  border-bottom: 1px solid
+    ${({ theme }) => `${theme.colors.mainHiglightColorInverted}1f`};
+  border-radius: 1px;
+
+  ${({ checked }) =>
+    checked &&
+    css`
+      > * > :first-child {
+        background: ${({ theme }) => theme.colors.mainHiglightColor};
+      }
+      > :first-child > :first-child {
+        border-top-left-radius: 8px;
+        border-bottom-left-radius: 8px;
+      }
+      > :last-child > :first-child {
+        border-top-right-radius: 8px;
+        border-bottom-right-radius: 8px;
+      }
+
+      box-shadow: 0px 17px 80px rgba(0, 0, 0, 0.1);
+    `}
+  &:hover {
+    > * > :first-child {
+      background: ${({ theme }) => theme.colors.mainDimColor};
+    }
+    > :first-child > :first-child {
+      border-top-left-radius: 8px;
+      border-bottom-left-radius: 8px;
+    }
+    > :last-child > :first-child {
+      border-top-right-radius: 8px;
+      border-bottom-right-radius: 8px;
+    }
+
+    box-shadow: 0px 17px 80px rgba(0, 0, 0, 0.1);
+  }
+`;
+
+const Radio = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: center;
+  align-items: center;
+  width: 44px;
+  height: 44px;
+`;
+
+const TitleCellContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: start;
+  align-items: center;
+  gap: 4px;
+  cursor: pointer;
+`;
+
+const IconContainer = styled.div<{ enabled: boolean }>`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  margin-top: 2px;
+  width: 16px;
+  height: 16px;
+  fill: ${({ theme }) => theme.colors.buttonDimIconColor};
+  visibility: ${({ enabled }) => (enabled ? "visible" : "hidden")};
+`;
+
+type InsertLayerProps = {
+  onImport: (object: { name: string; url: string; token?: string }) => void;
+  onCancel: () => void;
+};
+
+export const ArcGisImportPanel = ({ onImport, onCancel }: InsertLayerProps) => {
+  const dispatch = useAppDispatch();
+  const arcGisContentArray = useAppSelector(selectArcGisContent);
+  const arcGisContentSelected = useAppSelector(selectArcGisContentSelected);
+  const sortAscending = useAppSelector(selectSortAscending);
+  const sortColumn = useAppSelector(selectSortColumn);
+  const loadingStatus = useAppSelector(selectStatus);
+
+  const isLoading = loadingStatus === "loading";
+
+  useEffect(() => {
+    dispatch(resetArcGisContentSelected());
+  }, []);
+
+  const handleImport = () => {
+    const arcGisItem = arcGisContentArray.find(
+      (item) => item.id === arcGisContentSelected
+    );
+    if (arcGisItem) {
+      onImport(arcGisItem);
+    }
+  };
+
+  const onSort = (dataColumnName: ArcGisContentColumnName) => {
+    if (sortColumn === dataColumnName) {
+      dispatch(setSortAscending(!sortAscending));
+    } else {
+      dispatch(setSortColumn(dataColumnName));
+    }
+  };
+
+  const formatDate = (date: number) => {
```

Comment:
`formatTimestamp` is a better name. Move it to `src\utils\format\format-utils.ts`

Reply:
Done.